assets/js/base/components/cart-checkout/totals/shipping/shipping-address.tsx

```diff
@@ -31,15 +28,17 @@ export const ShippingAddress = ( {
 	setIsShippingCalculatorOpen,
 	shippingAddress,
 }: ShippingAddressProps ): JSX.Element | null => {
-	const addressComplete = isAddressComplete( shippingAddress );
 	const { isEditor } = useEditorContext();
 	const prefersCollection = useSelect( ( select ) =>
 		select( CHECKOUT_STORE_KEY ).prefersCollection()
 	);
-	// If the address is incomplete, and we're not in the editor, don't show anything.
-	if ( ! addressComplete && ! isEditor ) {
+	const hasFormattedAddress = !! formatShippingAddress( shippingAddress );
+
+	// If we don't have formatted address, and we're not in the editor, don't show anything.
+	if ( ! hasFormattedAddress && ! isEditor ) {
```

Comment:
Yup, maybe the comment can explain _why_ `hasFormattedAddress` would be false? (i.e. the shopper hasn't entered one)

Reply:
@ralucaStan Thank for the suggestion, I will update the comment and add more testing steps.

>What case are we actually covering with this and why are we returning nothing?

If there is no default customer location in the store and the customer has not entered their address, but there is a default shipping method available for all locations, then we will hide the shipping calculator. This is because the store already has a default shipping address. This change was made in response to this request: https://github.com/woocommerce/woocommerce-blocks/pull/8141#issuecomment-1397239622

Here are the steps to reproduce this behavior:

1. Go to the WooCommerce settings page: `wp-admin/admin.php?page=wc-settings`.
3. Change 'Default customer location' to `No location by default`. 
4. Add any default(fallback) shipping rates. 
5. Open your site in a private session(incognito mode).
6. Add some products to the cart. 
7. Go to the cart page, and confirm the shipping calculator is hidden and default shipping rates are visible
<img width="841" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/11503784/8daa2400-7436-4463-a08c-95b87ec92eb6">

<img width="1566" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/11503784/0110b41a-29ee-4b4e-97f9-b76abe2f6b59">
<img width="756" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/11503784/c9372b61-170b-4a65-9e10-f2528023b8d3">

Reply:
I understand that we have this rather complex sutation because we choose to have two buttons, one named "change address" and one named "enter an address to view shipping rates"? Otherwise I don't understand why we need to check if a formatted address is present to begin with to see the calculator.

Reply:
@senadir Yes, it wouldn't make sense to show the 'Add address for shipping rates' link when default rates are already displayed. Similarly, we should not display the 'Change address' link without an address. Hence we decided to hide it in this condition.

Reply:
@senadir No, we don't show a calculator in that scenario. It will be addressed in the https://github.com/woocommerce/woocommerce-blocks/issues/10327 ticket.